### ISSUE TITLE
[stable/airflow] Missing Postgres and Redis Password in 3.0.6

### ISSUE THREAD

**Issue by grumpybavarian** [helm/charts#15896]
**Describe the bug**
Since the we updated from airflow 3.0.2 to airflow 3.0.6 we are experiencing issues with the jobs being stuck in `scheduled` but never proceeding to `queued` or `running`. Checking the logs of the workers revealed the following log message over and over again: 
```
192.168.0.10 - - [24/Jul/2019 15:05:05] "GET / HTTP/1.1" 404 -
[2019-07-24 15:05:22,413: ERROR/MainProcess] consumer: Cannot connect to redis://thymeflow-redis-master:6379/1: NOAUTH Authentication required..
Trying again in 20.00 seconds...
```
Checking the environment variables revealed that `REDIS_PASSWORD` and `POSTGRES_PASSWORD` were not set:
```
airflow@thymeflow-worker-0:~$ env | grep -i pass
airflow@thymeflow-worker-0:~$
```
in comparison these were set in 3.0.2:
```
airflow@thymeflow-worker-0:~$ env | grep -i pass
REDIS_PASSWORD=airflow
POSTGRES_PASSWORD=airflow
airflow@thymeflow-worker-0:~$
```

Here's the (shortened) scheduler yaml generated by deploying airflow 3.0.6 where you can see that the environment variables containing the redis and postgres password are missing:
```
# Source: airflow/templates/deployments-scheduler.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: thymeflow-scheduler
  labels:
    app: airflow
    component: scheduler
    chart: airflow-3.0.6
    release: thymeflow
    heritage: Tiller
spec:
  replicas: 1
  template:
    spec:
      containers:
        - name: airflow-scheduler
          image: gitlab.beno.ai:4567/bic/alchemy/thymeflow:223-local-airflow
          imagePullPolicy: Always
          envFrom:
          - configMapRef:
              name: "thymeflow-env"
          env:          
            - name: POSTGRES_USER
              value: "postgres"
          resources:
            limits:
              cpu: "2"
              memory: 2Gi
            requests:
              cpu: "0.5"
              memory: 256Mi
 
          args:
            - "bash"
            - "-c"
            - >
              echo 'waiting 10s...' &&
              sleep 10 &&
              mkdir -p /usr/local/airflow/.local/bin &&
              export PATH=/usr/local/airflow/.local/bin:$PATH &&
              echo "executing initdb" &&
              airflow initdb &&
              echo "adding variables" &&
              airflow variables -i /usr/local/variables-pools/variables.json &&
              echo "adding pools" &&
              airflow pool -i /usr/local/variables-pools/pools.json &&
              echo "executing scheduler" &&
              airflow scheduler -n -1
```
And here's the (shortened) scheduler yaml generated by deploying airflow 3.0.2:
```
# Source: airflow/templates/deployments-scheduler.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: thymeflow-scheduler
  labels:
    app: airflow
    component: scheduler
    chart: airflow-3.0.2
    release: thymeflow
    heritage: Tiller
spec:
  
    spec:
      containers:
        - name: airflow-scheduler
          image: gitlab.beno.ai:4567/bic/alchemy/thymeflow:223-local-airflow
          imagePullPolicy: Always
          envFrom:
          - configMapRef:
              name: "thymeflow-env"
          env:          
            - name: POSTGRES_USER
              value: "postgres"
            - name: POSTGRES_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: thymeflow-postgresql
                  key: postgres-password
            - name: REDIS_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: thymeflow-redis
                  key: redis-password
          resources:
            limits:
              cpu: "2"
              memory: 2Gi
            requests:
              cpu: "0.5"
              memory: 256Mi
            
          args:
            - "bash"
            - "-c"
            - >
              echo 'waiting 10s...' &&
              sleep 10 &&
              mkdir -p /usr/local/airflow/.local/bin &&
              export PATH=/usr/local/airflow/.local/bin:$PATH &&
              echo "executing initdb" &&
              airflow initdb &&
              echo "adding variables" &&
              airflow variables -i /usr/local/variables-pools/variables.json &&
              echo "adding pools" &&
              airflow pool -i /usr/local/variables-pools/pools.json &&
              echo "executing scheduler" &&
              airflow scheduler -n -1
```

**Version of Helm and Kubernetes**:
Helm: 2.14
Kubernetes: v1.13.4

**Which chart**:
stable/airflow

**What happened**:
Jobs were not being scheduled because workers could access the postgres db and the redis queue.

**What you expected to happen**:
Jobs being scheduled normally. Environment variables set as in 3.0.2

**How to reproduce it** (as minimally and precisely as possible):
deploy airflow 3.0.6 and try to schedule jobs. Also check the set environment variables in the helm manifest and in the scheduler and workers.

**Anything else we need to know**:

**Comment by jinbal** [comment]
Can confirm I'm experiencing the same issue - rolled back to 3.0.2 as a workaround

**Comment by jparklab** [comment]
I had the same issue, and fixed it by manually adding POSTGRES_PASSWORD and REDIS_PASSWORD environment variables from secrets.

**Comment by mohannadbanayosi** [comment]
It's because of a change made in chart v3.0.5 where you have to specify the `.postgresql.existingSecret` and `.redis.existingSecret`. It won't specify the secrets if you leave them empty. 

Reference: [[stable/airflow] update condition (#15712)](https://github.com/helm/charts/commit/27fb92ad0f7f3f520a1d0176ae275e86408ce557#diff-eea042f1e2f305963128c6564622bc76)

**Comment by vinayan3** [comment]
@mohannadbanayosi Any updates on getting your PR through? I'm also, hitting this.

**Comment by mohannadbanayosi** [comment]
Unfortunately not, but I will push for it again today.

**Comment by mohannadbanayosi** [comment]
@vinayan3 Unfortunately not, but I will push for it again today.

**Comment by mohannadbanayosi** [comment]
@vinayan3 it is merged now and `4.0.5` should contain the fix.
